storage/postgres/storage.sql

```diff
@@ -0,0 +1,138 @@
+-- render the data in the tree unusable or inconsistent.
+CREATE TABLE IF NOT EXISTS trees ( 
+  tree_id                  BIGINT NOT NULL, 
+  tree_state               E_TREE_STATE NOT NULL, 
+  tree_type                E_TREE_TYPE NOT NULL, 
+  hash_strategy            E_HASH_STRATEGY NOT NULL, 
+  hash_algorithm           E_HASH_ALGORITHM NOT NULL, 
+  signature_algorithm      E_SIGNATURE_ALGORITHM NOT NULL, 
+  display_name             VARCHAR(20), 
+  description              VARCHAR(200), 
+  create_time_millis       BIGINT NOT NULL, 
+  update_time_millis       BIGINT NOT NULL, 
+  max_root_duration_millis BIGINT NOT NULL, 
+  private_key              BYTEA NOT NULL, 
+  public_key               BYTEA NOT NULL, 
+  deleted                  BOOLEAN, 
+  delete_time_millis       BIGINT, 
+  PRIMARY KEY(tree_id) 
+); 
+
+-- This table contains tree parameters that can be changed at runtime such as for
+-- administrative purposes.
+CREATE TABLE IF NOT EXISTS tree_control(
+  tree_id                   BIGINT NOT NULL,
+  signing_enabled           BOOLEAN NOT NULL,
+  sequencing_enabled        BOOLEAN NOT NULL,
+  sequence_interval_seconds INTEGER NOT NULL,
+  PRIMARY KEY(tree_id),
+  FOREIGN KEY(tree_id) REFERENCES trees(tree_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS subtree(
+  tree_id               BIGINT NOT NULL,
+  subtree_id            BIT VARYING(255) NOT NULL,
+  nodes                 BYTEA NOT NULL,
+  subtree_revision      INTEGER NOT NULL, 
+  PRIMARY KEY(tree_id, subtree_id, subtree_revision),
+  FOREIGN KEY(tree_id) REFERENCES Trees(tree_id) ON DELETE CASCADE
+);
+
+-- The TreeRevisionIdx is used to enforce that there is only one STH at any
+-- tree revision
+CREATE TABLE IF NOT EXISTS tree_head(
+  tree_id                BIGINT NOT NULL,
+  tree_head_timestamp    BIGINT,
+  tree_size              BIGINT,
+  root_hash              BIT VARYING(255) NOT NULL,
+  root_signature         BIT VARYING(1024) NOT NULL,
+  tree_revision          BIGINT,
+  PRIMARY KEY(tree_id, tree_head_timestamp),
+  FOREIGN KEY(tree_id) REFERENCES trees(tree_id) ON DELETE CASCADE
+);
+
+CREATE UNIQUE INDEX TreeHeadRevisionIdx
+  ON tree_head(tree_id, tree_revision);
+
+-- ---------------------------------------------
+-- Log specific stuff here
+-- ---------------------------------------------
+
+-- Creating index at same time as table allows some storage engines to better
+-- optimize physical storage layout. Most engines allow multiple nulls in a
+-- unique index but some may not.
+
+-- A leaf that has not been sequenced has a row in this table. If duplicate leaves
+-- are allowed they will all reference this row.
+CREATE TABLE IF NOT EXISTS leaf_data(
+  tree_id               BIGINT NOT NULL,
+  -- This is a personality specific has of some subset of the leaf data.
+  -- It's only purpose is to allow Trillian to identify duplicate entries in
+  -- the context of the personality.
+  leaf_identity_hash     BIT VARYING(255) NOT NULL,
```

Comment:
should this be `BYTEA` or `BYTEA(32)`?

Reply:
In other storage schemas, we've deliberately kept the field size overly large in order to accommodate > 256bit hashes for future expansion.